server/src/controllers/workspace.js

```diff
@@ -0,0 +1,249 @@
+import Logger from '../logger';
+import Location from '../models/location';
+import Workspace from '../models/workspace';
+
+const logger = Logger('controller > workspace');
+
+exports.getWorkspaces = async (request, response, next) => {
+  const { qLocation, page = 1, count = 10 } = request.params;
```

Comment:
too much logic in one function. 
Break thr function into simpler components

Reply:
Done!